SerhatTan/hw3/process.js

```diff
@@ -0,0 +1,62 @@
+//Listening the some dom elements and fetch their values
+//Calling operations function from Calculator class
+//Sending values to operations function parameters
+//Clear screen with click function event
+
+
+
+var classname = document.getElementsByClassName("btn");
+var digits = [];
+
+digitProcess = function(e) {
+	// e.target.innerHTML is the button's number like 5, 6 
+	digits.push(e.target.innerHTML);
+	var number = digits.join("");
+	document.getElementById("screen").innerHTML = number;
+};
+
+calculate = function() {
+	var screen = document.getElementById("screen");
+	data = screen.innerHTML;
+
+	console.log(data.indexOf("+"));
+	// Gets the first index where a operand occours
+	if (data.indexOf("+")!=-1) {
+		var index = data.indexOf("+");
+		operation = "sum";
+	} else if (data.indexOf("-")!=-1) {
+		var index = data.indexOf("-");
+		operation = "subtraction";
+	} else if (data.indexOf("*")!=-1) {
+		var index = data.indexOf("*");
+		operation = "multiply";
+	} else if (data.indexOf("/")!=-1) {
+		var index = data.indexOf("/");
+		operation = "division";
+	}
+	var number1 = data.substr(0, index); // Gets the first part
+	var number2 = data.substr(index + 1).slice(0, -1); // Gets the second part
+
+	var calculator = new Calculator();
+	result = calculator[operation](number1, number2);
+	screen.innerHTML = result;
+	digits = [];
+	digits.push(result);

```

Comment:
`	result = calculator[operation](number1, number2);
`

Kullanıcının yapmak istediği işlemi bir üstteki if else bloklarında algılayıp bunu değişken olarak calculator sınıfında çağırmak istediğim method adı olarak yolluyorum number1 ve number2 parametreleri ile. Bir sınıftaki methodu çağırmak için ve bu methodun adını parametre olarak tanımlayabilmek için sınıfımızın adından sonra çağırmak istediğimiz methodu [ ] içerisine yazmamız gerekiyor o yüzden bu şekilde kullandım.